### DISCUSSION TITLE
How to show incoming request HTTP method and path

### DISCUSSION THREAD

**Discussion by ilmarivacklinsc** [aws-powertools/powertools-lambda-typescript#2257] [Q&A] [answered]
Is there a way to get the Traces list in CloudWatch show the incoming request HTTP method and path? This lambda is called via a Lambda function URL from Cloudfront, and we're using the middy middleware to wrap the handler as per [the documentation](https://docs.powertools.aws.dev/lambda/typescript/2.0.3/core/tracer/#lambda-handler).

<img width="1301" alt="Screenshot 2024-03-20 at 9 52 43" src="https://github.com/aws-powertools/powertools-lambda-typescript/assets/150438145/babeffd9-9c00-44b5-91e2-118d84066b87">

**Comment by dreamorosi** [comment, marked as answer]
Hi @ilmarivacklinsc, we had a similar request in the past and the feature request is tracked here #1179.

The short answer is that no, at the moment this is not possible because - if my memory serves me right - the base implementation of the `Segment` & `Subsegment` objects doesn't include methods that allow to add this data to them.

I think it should be possible however to add this. 

If possible I'd like to ask you to please add a 👍 to the linked issue, we usually prioritize our backlog based on demand. Alternatively, we also accept contributions so if you or anyone from the community would like to work on this I'd be happy to collaborate on a design and provide timely reviews to get it merged.
